pkg/platform/runtime/buildscript/json.go

```diff
@@ -90,48 +83,74 @@ func (f *FuncCall) MarshalJSON() ([]byte, error) {
 }
 
 func marshalReq(args []*Value) ([]byte, error) {
-	mArgs := make(map[string]interface{})
-	for _, argument := range args {
-		switch {
-		case argument.Assignment != nil:
-			if argument.Assignment.Key == buildexpression.RequirementNameKey && argument.Assignment.Value.Str != nil {
-				name, namespace := separateNamespace(*argument.Assignment.Value.Str)
-				mArgs[buildexpression.RequirementNameKey] = name
-				mArgs[buildexpression.RequirementNamespaceKey] = namespace
-			} else if argument.Assignment.Key == buildexpression.RequirementVersionKey && argument.Assignment.Value.Str != nil {
-				value := strings.Trim(*argument.Assignment.Value.Str, `"`)
-				versionReqs, err := model.VersionStringToRequirements(value)
-				if err != nil {
-					return nil, errs.Wrap(err, "Could not parse version requirements")
-				}
+	requirement := make(map[string]interface{})
 
-				var scriptReqs []*Value
-				for _, req := range versionReqs {
-					var assignments []*Assignment
-					comparator, ok := req[bpModel.VersionRequirementComparatorKey]
-					if ok {
-						assignments = append(assignments, &Assignment{buildexpression.RequirementComparatorKey, &Value{Str: ptr.To(comparator)}})
-					}
+	for _, arg := range args {
+		assignment := arg.Assignment
+		if assignment == nil {
+			return nil, errors.New(fmt.Sprintf("Cannot marshal %v", arg))
+		}
 
-					version, ok := req[bpModel.VersionRequirementVersionKey]
-					if ok {
-						assignments = append(assignments, &Assignment{buildexpression.RequirementVersionKey, &Value{Str: ptr.To(version)}})
+		switch {
+		// Marshal the name argument (e.g. name = "<namespace>/<name>") into
+		// {"name": "<name>", "namespace": "<namespace>"}
+		case assignment.Key == buildexpression.RequirementNameKey && assignment.Value.Str != nil:
+			name, namespace := separateNamespace(*assignment.Value.Str)
+			requirement[buildexpression.RequirementNameKey] = name
+			requirement[buildexpression.RequirementNamespaceKey] = namespace
+
+		// Marshal the version argument (e.g. version = <op>("<version>")) into
+		// {"version_requirements": [{"comparator": "<op>", "version": "<version>"}]}
+		case assignment.Key == buildexpression.RequirementVersionKey && assignment.Value.FuncCall != nil:
+			var requirements []*Value
+			var addRequirement func(*FuncCall) error // recursive function for adding to requirements list
+			addRequirement = func(funcCall *FuncCall) error {
+				switch name := funcCall.Name; name {
+				case eqFuncName:
+					fallthrough
+				case neFuncName:
+					fallthrough
+				case gtFuncName:
+					fallthrough
+				case gteFuncName:
+					fallthrough
+				case ltFuncName:
+					fallthrough
+				case lteFuncName:
```

Comment:
Couldn't we list all of these values for a single case rather than use `fallthrough` here? So it would be something like:

```
case eqFuncName, neFuncName, gtFuncName, etc:
   ...
```